internal/provider/resource_privatelink.go

```diff
@@ -113,9 +113,36 @@ func (r *PrivateLinkResource) Create(ctx context.Context, req resource.CreateReq
 		return
 	}
 
-	pl, err := r.client.CreatePrivateLinkAwait(ctx, nsID, apigen.PostPrivateLinkRequestBody{
-		ConnectionName: data.ConnectionName.ValueString(),
-		Target:         data.Target.ValueString(),
+	var (
+		connectionName = data.ConnectionName.ValueString()
+		target         = data.Target.ValueString()
+	)
+
+	pl, err := r.client.GetPrivateLinkByName(ctx, connectionName)
+	if err != nil {
+		// ignore not found error
+		if !errors.Is(err, cloudsdk.ErrPrivateLinkNotFound) {
+			resp.Diagnostics.AddError("Get failed", err.Error())
+			return
+		}
```

Comment:
If the Get call failed due to reasons like timeout, does it really make sense to continue to the creation? 🤔

Reply:
a general retry mechanism should be introduced later to fix problems like this.

Reply:
Even if we have try, my point is more of: should we simply stop and throw error or should we continue with creation when we encountered an unknown error? I'm leaning towards the former.

Reply:
yes, the current implementation is just throwing an error and stopping the whole procedure.

Reply:
Hmm, did I read something wrong? This is the current implementation with some unrelated code removed:

```go
pl, err := r.client.GetPrivateLinkByName(ctx, connectionName)
if err != nil {
		// Ignore returning errors that signify the resource is no longer existent
		if !errors.Is(err, cloudsdk.ErrClusterNotFound) {
			...
			return
		}
	} else {
		// None error case
	}

// If GetPrivateLinkByName throws a error that is not 404, we reach here
pl, err = r.client.CreatePrivateLinkAwait(...)

```

Looks like we simply proceed with creation the we encountered an unknown error at the Get call?

Reply:
as you can see here
```go
if !errors.Is(err, cloudsdk.ErrClusterNotFound) {
  return
}
```
we simply return, not carry on. 

Here is the logic I tend to implement:

if the error is `NotFound`, then proceed.
if the error is not `NotFound`, then return (abort)
if the error is `nil` (meaning there is an existing resource)
 1. delete the resource then proceed if the resource is in failed state
 2. return (abort) as a resource with the same already exist

Reply:
Oh, I missed the negate operation, the logic is perfectly fine in this case. Apology for the confusion🥹

Reply:
nah thanks for the thoughtful review, I'm making it clear later